services/order.py

```diff
@@ -0,0 +1,29 @@
+from django.contrib.auth import get_user_model
+
+from db.models import Order, Ticket, MovieSession
+from django.db import transaction
+
+
+@transaction.atomic
+def create_order(
+    tickets: list[dict], username: str, date: str | None = None
+) -> None:
+    user = get_user_model().objects.get(username=username)
+    order = Order.objects.create(user=user)
+    if date:
+        order.created_at = date
+        order.save()
+    for ticket in tickets:
+        movie_session = MovieSession.objects.get(id=ticket["movie_session"])
+        Ticket.objects.create(
+            order=order,
+            row=ticket["row"],
+            seat=ticket["seat"],
+            movie_session=movie_session,
+        )
+
+
+def get_orders(username: str | None = None) -> list:
```

Comment:
The type hint `str | None` is used here, which is only supported in Python 3.10 and later. Ensure that your environment supports this syntax or use `Optional[str]` from `typing` for compatibility with earlier versions.

Reply:
agree